client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx

```diff
@@ -61,11 +61,45 @@ export default function NotificationSettings( {
 	const [ isAddEmailModalOpen, setIsAddEmailModalOpen ] = useState< boolean >( false );
 	const [ selectedEmail, setSelectedEmail ] = useState< StateMonitorSettingsEmail | undefined >();
 	const [ selectedAction, setSelectedAction ] = useState< AllowedMonitorContactActions >();
+	const [ initialSettings, setInitialSettings ] = useState< InitialMonitorSettings >( {
+		enableEmailNotification: false,
+		enableMobileNotification: false,
+		selectedDuration: defaultDuration,
+		emailContacts: [],
+	} );
+	const [ hasUnsavedChanges, setHasUnsavedChanges ] = useState< boolean >( false );
 
 	const isMultipleEmailEnabled = isEnabled(
 		'jetpack/pro-dashboard-monitor-multiple-email-recipients'
 	);
 
+	// Check if any unsaved changes are present and prompt user to confirm before closing the modal.
+	const handleOnClose = useCallback( () => {
+		const unsavedChangesExist =
```

Comment:
It's good that we are now tracking changes. Maybe we can move `unsavedChangesExist` outside this callback and use it to disable the 'Save' button at the same time if this is **false**. wdyt?

Reply:
Agree. I have made the changes - https://github.com/Automattic/wp-calypso/pull/77485/commits/416321f7be7c9a159e078a1dddf2569973d2e703